### PR TITLE
Feature/gh45 profile load

### PR DESCRIPTION
**Pull request by Ulrond**
- #45 - renamed assert to profile, corrected testing suites.
- corrected memory leak from string function
- added extra error return codes for NULL_PARAM
-

### CODE REVIEW

File: include/ut_kvp_profile.h [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+#define UT_ASSERT_EQUAL_KVP_PROFILE_UINT64(checkValue, key) UT_ASSERT_EQUAL(ut_kvp_getUInt64Field(ut_kvp_profile_getInstance(), key), checkValue); 
+
+/**! Asserts that a string KVP field matches the expected value. */
+#define UT_ASSERT_EQUAL_KVP_PROFILE_STRING(checkValue, key) \
+    {                                                                       \
+        char result_kvp[UT_KVP_MAX_ELEMENT_SIZE]={0};                        \
+        ut_kvp_status_t status;                 \
+        status = ut_kvp_getStringField(ut_kvp_profile_getInstance(), key, result_kvp, UT_KVP_MAX_ELEMENT_SIZE);  \
+        UT_ASSERT( status == UT_KVP_STATUS_SUCCESS ); \
+        if ( status != UT_KVP_STATUS_SUCCESS ) \
```
Comment (Ulrond) [inline include/ut_kvp_profile.h:67]: needs to be == UK_KVP_STATUS_SUCCESS

File: include/ut_kvp_profile.h [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        {\
+            UT_ASSERT_STRING_EQUAL(checkValue, result_kvp);                       \
+        }                                                                   \
```
Comment (Ulrond) [inline include/ut_kvp_profile.h:70]: add else { UT_ASSERT( TRUE } );

**Review by Ulrond — COMMENTED** [review]
added comments.

File: tests/src/ut_test_kvp_profile.c [hunk 3]
```diff
@@ -36,37 +36,61 @@ static UT_test_suite_t *gpAssertSuite1 = NULL;
 static UT_test_suite_t *gpAssertSuite2 = NULL;
 static UT_test_suite_t *gpAssertSuite3 = NULL;
 
+void test_ut_kvp_profile_uint8(void)
+{
+    uint32_t checkField = 0xdeadbeef;
```
Comment (kanjoe24) [inline tests/src/ut_test_kvp_profile.c]: checkField value needs to be updated.

File: tests/src/ut_test_kvp_profile.c [hunk 4]
```diff
@@ -36,37 +36,61 @@ static UT_test_suite_t *gpAssertSuite1 = NULL;
...
+    UT_LOG_STEP( "test_ut_kvp_profile_uint8 - start" );
+    UT_ASSERT_KVP_EQUAL_PROFILE_UINT8( checkField, "decodeTest/checkUint8IsDeDec" );
+    UT_ASSERT_KVP_EQUAL_PROFILE_UINT8( checkField, "decodeTest/checkUint8IsDeHex" );
+    UT_LOG_STEP( "test_ut_kvp_profile_uint8 - end" );
+}
+
+void test_ut_kvp_profile_uint16(void)
+{
+    uint32_t checkField = 0xdeadbeef;
```
Comment (kanjoe24) [inline tests/src/ut_test_kvp_profile.c]: checkField value needs to be updated.

**Review by kanjoe24 — COMMENTED** [review]
checkField value needs to be updated.

**Review by kanjoe24 — APPROVED** [review]
Looks good to me

**Pull request closed without merging**